### ISSUE TITLE
Build Error on Python 3.11

### ISSUE THREAD

**Issue by WilliamJamieson** [asdf-format/sphinx-asdf#74]
When I make the `asdf-astropy` docs locally in a Python 3.11 virtual env I consistently get the following error:
```
Extension error (sphinx_asdf.connections):
Handler <function autogenerate_schema_docs at 0x131564f40> for event 'builder-inited' threw an exception (exception: global flags not at the start of the expression at position 441)
```

I can reproduce this consistently (across newly created clean envs).

`pip list`:
```
Package                       Version              Editable project location
----------------------------- -------------------- -----------------------------------------------------
alabaster                     0.7.13
asdf                          2.15.0
asdf-astropy                  0.4.1.dev25+gc6608be /Users/wjamieson/workspaces/asdf-astropy/asdf-astropy
asdf-coordinates-schemas      0.2.0
asdf-standard                 1.0.3
asdf-transform-schemas        0.3.0
asdf-unit-schemas             0.1.0
astropy                       5.3.1
astropy-sphinx-theme          1.1
attrs                         23.1.0
Babel                         2.12.1
certifi                       2023.5.7
charset-normalizer            3.2.0
contourpy                     1.1.0
coverage                      7.2.7
cycler                        0.11.0
docutils                      0.18.1
fonttools                     4.41.0
graphviz                      0.20.1
hypothesis                    6.81.1
idna                          3.4
imagesize                     1.4.1
importlib-metadata            6.8.0
iniconfig                     2.0.0
Jinja2                        3.1.2
jmespath                      1.0.1
jsonschema                    4.17.3
kiwisolver                    1.4.4
MarkupSafe                    2.1.3
matplotlib                    3.7.2
mistune                       3.0.1
numpy                         1.25.1
numpydoc                      1.5.0
packaging                     23.1
Pillow                        10.0.0
pip                           23.1.2
pluggy                        1.2.0
psutil                        5.9.5
pyerfa                        2.0.0.3
Pygments                      2.15.1
pyparsing                     3.0.9
pyrsistent                    0.19.3
pytest                        7.4.0
pytest-arraydiff              0.5.0
pytest-astropy                0.10.0
pytest-astropy-header         0.2.2
pytest-cov                    4.1.0
pytest-doctestplus            0.13.0
pytest-filter-subpackage      0.1.2
pytest-mock                   3.11.1
pytest-openfiles              0.5.0
pytest-remotedata             0.4.0
python-dateutil               2.8.2
PyYAML                        6.0
requests                      2.31.0
scipy                         1.11.1
semantic-version              2.10.0
setuptools                    65.5.0
six                           1.16.0
snowballstemmer               2.2.0
sortedcontainers              2.4.0
Sphinx                        6.2.1
sphinx-asdf                   0.2.0
sphinx-astropy                1.9.1
sphinx-automodapi             0.15.0
sphinx-bootstrap-theme        0.8.1
sphinx-gallery                0.13.0
sphinx-rtd-theme              1.2.2
sphinxcontrib-applehelp       1.0.4
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        2.0.1
sphinxcontrib-jquery          4.1
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.5
toml                          0.10.2
tomli                         2.0.1
urllib3                       2.0.3
zipp                          3.16.1
```

**Comment by braingram** [comment]
Thanks for opening the issue. Would you be interested in reviewing #72?

**Comment by braingram** [comment]
Fixed in #72
